lib/carbon/util.py

```diff
@@ -298,3 +315,71 @@ def __init__(classObj, name, bases, members):
     super(PluginRegistrar, classObj).__init__(name, bases, members)
     if hasattr(classObj, 'plugin_name'):
       classObj.plugins[classObj.plugin_name] = classObj
+
+
+class TaggedSeries(object):
+  @classmethod
+  def parse(cls, path):
+    tags = {}
+
+    # if path is in openmetrics format: metric{tag="value",...}
+    # https://github.com/RichiH/OpenMetrics
+    if path[-2:] == '"}' and '{' in path:
+      (metric, rawtags) = path[0:-1].split('{', 2)
+      if not metric:
+        raise Exception('Cannot parse path %s, no metric found' % path)
+
+      while len(rawtags) > 0:
+        m = re.match(r'([^=]+)="((?:[\\]["\\]|[^"\\])+)"(:?,|$)', rawtags)
+        if not m:
+          raise Exception('Cannot parse path %s, invalid segment %s' % (path, rawtags))
+
+        tags[m.group(1)] = m.group(2).replace(r'\"', '"').replace(r'\\', '\\')
+        rawtags = rawtags[len(m.group(0)):]
+
+      tags['name'] = metric
+      return cls(metric, tags)
+
+    # path is a carbon path with optional tags: metric;tag=value;...
+    segments = path.split(';')
+
+    metric = segments[0]
+    if not metric:
+      raise Exception('Cannot parse path %s, no metric found' % path)
+
+    for segment in segments[1:]:
+      tag = segment.split('=', 1)
+      if len(tag) != 2 or not tag[0]:
+        raise Exception('Cannot parse path %s, invalid segment %s' % (path, segment))
+
+      tags[tag[0]] = tag[1]
+
+    tags['name'] = metric
+    return cls(metric, tags)
+
+  @staticmethod
+  def format(tags):
+    return tags.get('name', '') + ''.join(sorted([
+      ';%s=%s' % (tag, value)
+      for tag, value in tags.items()
+      if tag != 'name'
+    ]))
+
+  @staticmethod
+  def encode(metric, sep='.'):
+    # if metric is tagged, encode it for storage in whisper etc
```

Comment:
Add an example of such encoding here ? (to make it clearer what the function does)